src/new_pass.c

```diff
@@ -1,92 +1,106 @@
+    if(fp == NULL)
+    {
+        fp = fopen("EMP.DAT","wb+");
+        if(fp == NULL)
+        {
+            printf("Connot open file");
+            exit(1);
         }
-        printf("*");
     }
+        printf("*");
+        }
 
-    if (strcmp(e.pass, pass) == 0) {
-        for (i = 0; i <= 10; i++) {
+    if (strcmp(e.pass,pass) == 0){
+        for(i=0;i<=10;i++){
             pass[i] = '\0';
         }
-        printf("\nPlease enter your new password : ");
-        for (i = 0; i < 10; i++) {
-            c = getch();
-            if ((c == 13) || (c == 10) || (c == 32)) {
-                break;
+    printf("\n\nPlease enter your new password : ");
+    for (i=0;i<10;i++){
+        c=getch();
+        if ((c== 13) ||(c == 10)||(c == 32)){
+            break;
+        }
+        pass[i] = c;
+        printf("*");
+        }
+    printf("\nPlease re-enter your new password : ");
+    for (i=0;i<10;i++){
+        c=getch();
+        if ((c== 13) ||(c == 10)||(c == 32)){
+            break;
+        }
+        npass[i] = c;
+        printf("*");
+    }
+
+    if (strcmp(pass,npass) == 0){
+        fp = fopen("EMP.DAT","rb+");
+        if(fp == NULL){
+            fp = fopen("EMP.DAT","wb+");
+            if(fp == NULL){
+                printf("Connot open file");
+                exit(1);
             }
-            pass[i] = c;
-            printf("*");
         }
-        printf("\nPlease re-enter your new password : ");
-        for (i = 0; i < 10; i++) {
-            c = getch();
-            if ((c == 13) || (c == 10) || (c == 32)) {
-                break;
+        /* test code below */
+        fclose(fp);
+        fp = fopen("EMP.DAT","rb+");
+        if(fp == NULL){
+            fp = fopen("EMP.DAT","wb+");
+            if(fp == NULL){
+                printf("Connot open file");
+                exit(1);
             }
-            npass[i] = c;
-            printf("*");
         }
 
-        if (strcmp(pass, npass) == 0) {
-            fp = fopen("EMP.DAT", "rb+");
-            if (fp == NULL) {
-                fp = fopen("EMP.DAT", "wb+");
-                if (fp == NULL) {
-                    printf("Connot open file");
-                    exit(1);
-                }
-            }
-            rewind(fp);
-            while (fread( & e, recsize, 1, fp) == 1) {
-                if (e.id == user) {
-                    strcpy(e.pass, npass);
-                    fseek(fp, -recsize, SEEK_CUR); /* move the cursor 1 step back from current position */
-                    fwrite( & e, recsize, 1, fp);
-                    printf("\n\nPassword successfully chnaged");
-                    getch();
-                    break;
-                }
+        // test code ends
+        rewind(fp);
+        while(fread(&e,recsize,1,fp)==1){
+            if(e.id == user){
+                strcpy(e.pass,npass);
+                fseek(fp,-recsize,SEEK_CUR); /*move the cursor 1 step back from current position */

```

Comment:
handle the error case if it returns -1

Reply:
the fseek function moves the file pointer to the desired location and  it returns zero if successful
and non zero if it is not successful and it is handled by a separate file.

Reply:
okay.